src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp

```diff
@@ -224,6 +224,9 @@ CHIP_ERROR DefaultOTARequestorDriver::UpdateNotFound(UpdateNotFoundReason reason
 
 void DefaultOTARequestorDriver::UpdateDownloaded()
 {
+    // Download complete so reset provider retry counter
+    mProviderRetryCount = 0;
```

Comment:
I'm wondering if it's cleaner to just reset in `DefaultOTARequestorDriver::HandleIdleStateEnter`? That's where you reset the other retry counter and if we ever enter the Idle state, that means we're definitely done with retrying, e.g. in the case of Busy, the new state is DelayedOnQuery and not idle.

Reply:
Since `mProverRetryCount` is incremented in `DefaultOTARequestorDriver::SendQueryImage()`, I think for parity it is clearer to reset it in `UpdateDownloaded()`.  There could be various reasons why OTA-R enters idle state, such as watchdog timer kicking in due to no progress.

Reply:
@carol-apple, made the changes in `CustomOTARequestorDriver::UpdateDownloaded()` as we discussed.  So now, we reset `mProverRetryCount` in `CustomOTARequestorDriver::UpdateDownloaded()` in the case where we do not apply update. And in the case we do apply update, all member variables will be reset when the image is reloaded anyways.  This is to take care of the case where Apply Update fails and we want to count it against `mProverRetryCount`.